velox/functions/sparksql/Arithmetic.h

```diff
@@ -351,6 +351,45 @@ struct ToHexBigintFunction {
   }
 };
 
+template <typename T>
+struct WidthBucketFunction {
+  FOLLY_ALWAYS_INLINE bool call(
+      int64_t& result,
+      double operand,
+      double bound1,
+      double bound2,
+      int64_t bucketCount) {
+    if (bucketCount <= 0 ||
+        bucketCount == std::numeric_limits<long long>::max() ||
+        std::isnan(operand) || bound1 == bound2 || std::isnan(bound1) ||
+        std::isinf(bound1) || std::isnan(bound2) || std::isinf(bound2)) {
+      return false;
+    }
+
+    double lower = std::min(bound1, bound2);
+    double upper = std::max(bound1, bound2);
+
+    if (operand < lower) {
+      result = 0;
+    } else if (operand > upper) {
+      VELOX_USER_CHECK_NE(
+          bucketCount,
+          std::numeric_limits<int64_t>::max(),
```

Comment:
Spark return null when the bucketCount is max long, you could try `__builtin_add_overflow`
```
spark-sql (default)> select width_bucket(5.3, 0.2, 10.6, 9223372036854775807);
NULL
Time taken: 0.136 seconds, Fetched 1 row(s)
spark-sql (default)> set spark.sql.ansi.enabled=false
                   > ;
spark.sql.ansi.enabled  false
Time taken: 0.057 seconds, Fetched 1 row(s)
spark-sql (default)> select width_bucket(5.3, 0.2, 10.6, 9223372036854775807);
NULL
Time taken: 0.099 seconds, Fetched 1 row(s)
spark-sql (default)> set spark.sql.ansi.enabled=true;
spark.sql.ansi.enabled  true
Time taken: 0.036 seconds, Fetched 1 row(s)
spark-sql (default)> select width_bucket(5.3, 0.2, 10.6, 9223372036854775807);
NULL
```

Reply:
add a test case to guard this